spec/forms/user_piv_cac_setup_form_spec.rb

```diff
@@ -103,6 +103,38 @@
       end
     end
 
+    context 'when piv cac is required' do
+      let(:token) { 'good-token' }
+      let(:x509_dn_uuid) { 'random-uuid-for-x509-subject' }
+      let(:form) do
+        described_class.new(user: user, token: token, nonce: nonce, name: 'Card 1',
+                            piv_cac_required: true)
+      end
+
+      it 'returns FormResponse with success: true when the token has an eku' do
+        allow(PivCacService).to receive(:decode_token).with(token) { eku_token_response(true) }
+
+        result = instance_double(FormResponse)
+        extra = { multi_factor_auth_method: 'piv_cac' }
+
+        expect(FormResponse).to receive(:new).
+          with(success: true, errors: {}, extra: extra).and_return(result)
+        expect(form.submit).to eq result
+        expect(form.error_type).to be_nil
+      end
+
+      it 'returns FormResponse with success: false when the token does not have an eku' do
+        allow(PivCacService).to receive(:decode_token).with(token) { eku_token_response(false) }
+
+        extra = { multi_factor_auth_method: 'piv_cac', key_id: 'foo' }
+
+        result = form.submit
+        expect(result.success?).to eq(false)
+        expect(result.errors).to eq(type: 'certificate.not_auth_cert')
+        expect(result.extra).to eq(extra)
+      end
```

Comment:
I was thinking more like.... (and also getting rid of stubbing like in the other test)
```suggestion 
      let(:eku_token_response) do
        { 'nonce' => nonce, 'key_id' => 'foo', 'has_eku' => has_eku, 'uuid' => x509_dn_uuid,
        'subject' => 'x509-subject' }
      end

      context 'when the token has an eku' do
        let(:has_eku) { true }
        it 'returns FormResponse with success: true ' do
          allow(PivCacService).to receive(:decode_token).with(token).and_return(eku_token_response)

          extra = { multi_factor_auth_method: 'piv_cac' }

          result = form.submit
          expect(result.success?).to eq(true)
          expect(result.errors).to eq(type: 'certificate.not_auth_cert')
          expect(result.error_type).to be_nil
          expect(result.extra).to eq(extra)
        end

      context 'when the token does not have an eku' do
        let(:has_eku) { false }

        it 'returns FormResponse with success: false' do
          allow(PivCacService).to receive(:decode_token).with(token).and_return(eku_token_response)

          extra = { multi_factor_auth_method: 'piv_cac', key_id: 'foo' }

          result = form.submit
          expect(result.success?).to eq(false)
          expect(result.errors).to eq(type: 'certificate.not_auth_cert')
          expect(result.extra).to eq(extra)
        end
```